### ISSUE TITLE
May I ask for the final IoU value or pixelwise accuracy of ur model?

### ISSUE THREAD

**Issue by kallyshall** [log4422/Unet_voc2012#1]

**Comment by log4422** [comment]
Sorry for my late answer (Corona).
I´m acctually not finish yet.
 
I´ve tried end-to end trainig but it doesn´t work at all (mIoU about 12%, i have tried different loss fuctions e.g cross_entropy, tversky_loss)

After using (freezed) pre-trained parameters the mIoU is about (40%), but i think it could be much better if you unfreez the parameters after a couple of training epochs training. But i´m not doing it because this is just a (college) project with limited workload. 

The next few weeks i´m going to update this page. 
Hope i could help you a bit ;)

**Comment by kallyshall** [comment]
@log4422 Thank for ur reply
I am recently working on this unet-voc thing also as a college project, and I have done some experiment, hope may help or for further reference.

I have tried crossentropy, crossentropy with l1 norm, weighted crossentroy even IoU as the loss function. 
IoU as the loss function seems to gain the best IoU value, about 61%. But as every img in VOC has almost 50% background area, it turns out the network prefers to classify most pixels to the background class.
I also tried IoU loss without counting the background, it also has a bad performence when visualizing the seg result. So the crossentropy with l1 norm(can raise the pixelwise accuracy) wins a little.

Maybe I will try other pretrain weights other than vgg or other data augmention ways other than keras' img generator, but so far I think it's hard to get a good performence using purly unet/fcn + crossentropy loss.
